lib/src/code_generator/objc_interface.dart

```diff
@@ -209,35 +221,56 @@ class ObjCInterface extends BindingType {
       if (m.dartDoc != null) {
         s.write(makeDartDoc(m.dartDoc!));
       }
-      if (!isStatic && (superType?.hasMethod(m) ?? false)) {
-        s.write('  @override\n');
-      }
+
       s.write('  ');
-      if (isStatic) s.write('static ');
-      if (m.kind == ObjCMethodKind.propertySetter) {
-        s.write('set ');
+      if (isStatic) {
+        s.write('static ');
+
+        switch (m.kind) {
+          case ObjCMethodKind.method:
+            // static returnType methodName(NativeLibrary _lib, ...)
+            s.write(_getConvertedType(returnType, w, name));
+            s.write(' $methodName');
+            s.write(paramsToString(m.params, isStatic: true));
+            break;
+          case ObjCMethodKind.propertyGetter:
+            // static returnType getMethodName(NativeLibrary _lib)
+            s.write(_getConvertedType(returnType, w, name));
+            s.write(' get');
+            s.write(methodName[0].toUpperCase() + methodName.substring(1));
+            s.write(paramsToString([], isStatic: true));
+            break;
+          case ObjCMethodKind.propertySetter:
+            // static void setMethodName(NativeLibrary _lib, ...)
+            s.write(' void set');
```

Comment:
Nit: This setter should have a `returnType` of void, so you can change this to be consistent with this other cases, and that means you can pull the `s.write(_getConvertedType(returnType, w, name));` out the front where the static keyword is.

Similarly, the `paramsToString` call can be the same for all 3 cases, so can be moved below the switch.